openeogeotrellis/job_tracker_v2.py

```diff
@@ -85,48 +87,112 @@ class YarnAppReportParseException(Exception):
 class YarnStatusGetter(JobMetadataGetterInterface):
     """YARN app status getter"""
 
+    def get_application_url(self, application_id: str) -> str:
+        return url_join(
+            ConfigParams().yarn_rest_api_base_url,
+            f"/ws/v1/cluster/apps/{application_id}",
+        )
```

Comment:
Instead of grabbing the URL from the environment in this method, I would define a constructor for `YarnStatusGetter` that takes this URL and stores it in a member variable. This makes `YarnStatusGetter`'s dependency on that URL explicit.

Reply:
That's better indeed.
Done.